src/features/auth/useAuthState.tsx

```diff
@@ -0,0 +1,27 @@
+import {useEffect} from 'react'
+import { useDispatch } from 'react-redux'
+import { auth } from './Auth';
+import { login, logout } from '../user/userSlice';
+
+// Custom hook that sets up the auth state listener that's called whenever the auth state changes
+// onAuthStateChanged() will returns a User object if the user is logged in, otherwise it will return null
+const useAuthState = () => {
+    const dispatch = useDispatch();
+
+    useEffect(() => {
+        const unsubscribe = auth.onAuthStateChanged((loginUser) => {
+            if(loginUser) {
+                dispatch(login(loginUser.uid));
+            }
+            else {
+                dispatch(logout());
+            }
+        })
+        // cleanup function
+        return () => unsubscribe();
+    }, [dispatch]);
+
+    return;
+}
+
+export default useAuthState;
```

Comment:
How to handle user session? (Part 1)

- This useAuthState custom hook will set up the auth state listener that's called whenever the auth state changes. It's just monitoring the auth state.
- onAuthStateChanged() will returns a User object if the user is logged in, otherwise it will return null
- We can use this function and dispatch login() if User object is returned, or dispatch logout if null is returned.